src/rpc/mempool.cpp

```diff
@@ -344,10 +343,12 @@ UniValue MempoolToJSON(const CTxMemPool& pool, bool verbose, bool include_mempoo
         }
         LOCK(pool.cs);
         UniValue o(UniValue::VOBJ);
-        for (const CTxMemPoolEntry& e : pool.mapTx) {
-            const uint256& hash = e.GetTx().GetHash();
+        for (const auto& txinfo : pool.infoAll()) {
+            const uint256& hash = txinfo.tx->GetHash();
+            const auto entry{pool.GetEntry(Txid::FromUint256(hash))};
```

Comment:
nit 65c4b3bac0894dfdf7e1c756697f9dd0da1f97e7: don't need to go to uint256 and back again
```suggestion
            const Txid& hash = txinfo.tx->GetHash();
            const auto entry{pool.GetEntry(hash)};
```